### PR TITLE
Update !whatsdue

### PR DESCRIPTION
**Pull request by mitchmcdee**
Simplified formatting further and supported no course codes being entered (to throw error for user).

### CODE REVIEW

File: scripts/whatsdue.js [hunk 1]
```diff
@@ -97,9 +97,14 @@ module.exports = function (robot) {
 
     /**
      * Robot responds to a message containing `!whatsdue`.
-     * Requires at least one course.
      */
-    robot.respond(/!?whatsdue ((?:[a-z]{4}[0-9]{4} ?)+)/i, function (res) {
+    robot.respond(/!?whatsdue ?((?:[a-z]{4}[0-9]{4} ?)+)?/i, function (res) {
+        // If there are no matching groups, throw an error.
+        if (!res.match[1]) {
```
Comment (TRManderson) [inline scripts/whatsdue.js]: Is this the best check for no match?

Comment (mitchmcdee) [inline scripts/whatsdue.js]: res.match[1] is only populated by valid matches, if it doesn't exist; there are no matches. Is there another check you have in mind/can recommend?

Comment (TRManderson) [inline scripts/whatsdue.js]: `res.match.length`? I think I'm okay with it as is though

**Comment by TRManderson** [comment]
Let me know when you're happy for me to merge

File: scripts/whatsdue.js (hunk 1 above)
Comment (mitchmcdee) [inline scripts/whatsdue.js]: happy to change it if there is another compelling reason to :) but atm I think it reads better than if(res.match.length != 2) or whatever hehe

appreciate the comment Tom!

**Comment by mitchmcdee** [comment]
all good to go :)

**Pull request merged**